docker.js

```diff
@@ -0,0 +1,174 @@
+const _ = require("lodash");
+
+function createVolumeConfig(path) {
+  if (!path) {
+    throw new Error("Path is required to create Volume Config.");
+  }
+  if (!_.isString(path)) {
+    throw new Error("Path parameter must be a string.");
+  }
+
+  const pathEnvironmentVariableName = generateRandomEnvironmentVariableName();
+  const mountPointEnvironmentVariableName = generateRandomEnvironmentVariableName();
+  const mountPoint = generateRandomTemporaryPath();
+
+  return {
+    path: {
+      environmentVariable: {
+        name: pathEnvironmentVariableName,
+        value: path,
+      },
+      value: `$${pathEnvironmentVariableName}`,
+    },
+    mountPoint: {
+      environmentVariable: {
+        name: mountPointEnvironmentVariableName,
+        value: mountPoint,
+      },
+      value: `$${mountPointEnvironmentVariableName}`,
+    },
+  };
+}
+
+function sanitizeCommand(command) {
+  if (!command || !_.isString(command)) {
+    throw new Error("Command parameter must be a string.");
+  }
+
+  return `sh -c ${JSON.stringify(command)}`;
+}
+
+function mergeVolumeConfigsEnvironmentVariables(volumeConfigs) {
```

Comment:
Obtaining narrow set of information from a richer single source of information sounds more like `extract`ion rather than `merg(e)`ing, where multiple sources are expected.